src/index/scanners/default.rs

```diff
@@ -504,3 +362,80 @@ impl SearchBuilder for DefaultBuilder {
         }))
     }
 }
+
+fn seq_filter<T, F>(heap: BinaryHeap<T>, f: F) -> BinaryHeap<T>
+where
+    F: Fn(&T) -> bool,
+    T: Ord,
+{
+    match enable_prefilter() {
+        true => heap.into_iter().filter(f).collect(),
+        false => heap,
+    }
+}
```

Comment:
Why scan the entire heap before it's needed?

Reply:
Refactored with a Filter.